File: go/1094.go
package main

import "fmt"

func main() {
  var n, qtd, tot int
  var animal string
  cob := map[string]int{
    "C": 0,
    "R": 0,
    "S": 0,
  }

  fmt.Scanf("%d", &n)
  for i := 0; i < n; i++ {
    fmt.Scanf("%d", &qtd)   
    fmt.Scanf("%s", &animal)
    cob[animal] += qtd
    tot += qtd
  }

  fmt.Printf("Total: %d cobaias\n", tot)
  fmt.Printf("Total de coelhos: %d\n", cob["C"])
  fmt.Printf("Total de ratos: %d\n", cob["R"])
  fmt.Printf("Total de sapos: %d\n", cob["S"])
  fmt.Printf("Percentual de coelhos: %.2f %%\n", 100 * float64(cob["C"]) / float64(tot))
  fmt.Printf("Percentual de ratos: %.2f %%\n",   100 * float64(cob["R"]) / float64(tot))
  fmt.Printf("Percentual de sapos: %.2f %%\n",   100 * float64(cob["S"]) / float64(tot))
}

File: lua/1094.lua
local function split(inputstr)
  local sep = "%s"
  local t={}
  for str in string.gmatch(inputstr, "([^"..sep.."]+)") do
    table.insert(t, str)
  end
  return t
end
local n, qtd, tot = 0, 0, 0
local cob = {
  C = 0,
  R = 0,
  S = 0,
}

n = io.read()
local animal, line
for _ = 1, n do
  line = io.read()
  line = split(line)

  qtd = tonumber(line[1])
  animal = line[2]

  cob[animal] = cob[animal] + qtd
  tot = tot + qtd
end

print(string.format("Total: %d cobaias", tot))
print(string.format("Total de coelhos: %d", cob["C"]))
print(string.format("Total de ratos: %d", cob["R"]))
print(string.format("Total de sapos: %d", cob["S"]))
print(string.format("Percentual de coelhos: %.2f %%", 100 * cob["C"] / tot))
print(string.format("Percentual de ratos: %.2f %%", 100 * cob["R"] / tot))
print(string.format("Percentual de sapos: %.2f %%", 100 * cob["S"] / tot))
